test-outofproc/TriggerWithException.cs

```diff
@@ -23,14 +24,15 @@ public static class TriggerWithException
         public static void Run(
             [SqlTrigger("[dbo].[Products]", "SqlConnectionString")]
             IReadOnlyList<SqlChange<Product>> changes,
-            ILogger logger)
+            FunctionContext context)
         {
             if (!threwException)
             {
                 threwException = true;
                 throw new InvalidOperationException(ExceptionMessage);
             }
-            logger.LogInformation("SQL Changes: " + Utils.JsonSerializeObject(changes));
+            _loggerMessage(context.GetLogger("TriggerWithException"), "SQL Changes: " + Utils.JsonSerializeObject(changes), null);
```

Comment:
Is there a reason you're changing this?

Reply:
I keep seeing the exception "logger cannot be null" when I run it the other way:
<img width="1664" alt="image" src="https://user-images.githubusercontent.com/12754347/231859640-bc7733bc-e573-40d8-b7e5-c161266560d9.png">

Reply:
Ok, so it looks like using the execution context is just how you do it in OOP (for some reason). 

https://learn.microsoft.com/en-us/azure/azure-functions/dotnet-isolated-process-guide#logging

Could you simplify this though? It doesn't seem necessary to have this logger delegate. (if you're getting a compiler warning that's probably the one I've disabled for the rest of our test code as well - you should do the same here in the editorconfig)

Reply:
shouldn't we address the complier warning so users know how to overcome it instead of ignoring it?

Reply:
1. This is specifically test code, so it's not something users should really be using as a guide
2. It's normally a warning, we just by default treat everything as errors. It's not breaking to have it with the simplified version and the difference only really matters for things that are doing a high rate of logging which really care about performance (if this is the same warning that I'm thinking of)

Reply:
Will clean up the tests in a separate PR, thanks.